odoo/addons/base/models/ir_model.py

```diff
@@ -1491,6 +1491,55 @@ def group_names_with_access(self, model_name, access_mode):
         """ % access_mode, [model_name])
         return [('%s/%s' % x) if x[0] else x[1] for x in self._cr.fetchall()]
 
+    @tools.ormcache('self.env.uid', 'self.env.su')
```

Comment:
Should cache on self.env.su *or* self.env.uid, not the pair, because currently `(1, True)` and `(2, True)` will have two entries in the cache, even though they're the same "user".